packages/editor/src/components/post-actions/actions.js

```diff
@@ -161,186 +321,13 @@ function useTrashPostAction( postType ) {
 	);
 	return useMemo(
 		() => ( {
-			id: 'move-to-trash',
-			label: __( 'Move to Trash' ),
-			isPrimary: true,
-			icon: trash,
+			...trashPostAction,
 			isEligible( item ) {
```

Comment:
The problem is that the `isEligible` call here won't be re-called when the value of `canUser( 'delete', resource, item.id )` changes no?

Reply:
Yeah, maybe we need something like `useEligible` somewhere. Or maybe this object should define which capabilities to check.

Reply:
It will because we make the memo depend on canUserResolvers (details in https://github.com/WordPress/gutenberg/pull/62589#discussion_r1647524094) so when the resolver finishes a rerender happens and isEligible will be recalled.

Reply:
🤨 seems fine for now but not ideal

Reply:
It's very hard to follow. I'm not sure I understand what `canUserResolvers` is?

Reply:
canUserResolvers is the result getCachedResolvers().canUser. So basically when there is a change in the caching of canUser resolvers we will rerender making isEligible recalled. I will add some inline comments, explaining what we discussed.

Reply:
Also why keeping the action as object just to be used inside the `hook` action? Anyway since it's for 6.6, but it becomes clear that an action needs a way to have access to the store in more places than the callback and we should work towards that direction.

Reply:
@ntsekouras I did that to minimise the diff, it was huge and unclear what changed + less likely to have cherry pick conflicts, and if it does, easier to resolve manually (I don't know what has changed in these files recently)

Reply:
And it's useful for reusability https://github.com/WordPress/gutenberg/pull/62754#discussion_r1649305615